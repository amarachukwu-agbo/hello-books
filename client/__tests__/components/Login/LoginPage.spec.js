import React from 'react';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import ConnectedLoginPage, { LoginPage }
  from '../../../components/Login/LoginPage';

let wrapper;

describe('<LoginPage />', () => {
  beforeEach(() => {
    const props = {
      loginUser: () => Promise.resolve(),
      login: {
        isLoggingIn: false,
        isAuthenticated: false,
        user: null,
      },
    };
    wrapper = shallow(<LoginPage {...props} />);
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('calls submitForm()', () => {
    const values = {
      firstName: 'User',
      email: 'user@gmail.com',
    };
    sinon.spy(wrapper.instance(), 'submitForm');
    wrapper.instance().submitForm(values);
    expect(wrapper.instance().submitForm.calledOnce)
      .toEqual(true);
    expect(wrapper.instance().submitForm.calledWith(values));
  });
});

describe('<ConnectedLoginPage />', () => {
  it('should render successfully', () => {
    const mockStore = configureMockStore([thunk]);
    const store = mockStore({});
    const connectedWrapper = shallow(<ConnectedLoginPage store = {store}/>);
    expect(connectedWrapper.length).toBe(1);
  });
});
