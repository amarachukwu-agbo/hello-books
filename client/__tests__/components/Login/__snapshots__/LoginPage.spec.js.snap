// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginPage /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoginPage
    login={
      Object {
        "isAuthenticated": false,
        "isLoggingIn": false,
        "user": null,
      }
    }
    loginUser={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="wrapper valign-wrapper"
      >
        <div
          className="col s1 m2 l4 "
        />
        <div
          className="col s10 m7 l5"
        >
          <div
            className="row card-panel"
          >
            <div
              className="row"
            >
              <ReduxForm
                id="login-form"
                login={
                  Object {
                    "isAuthenticated": false,
                    "isLoggingIn": false,
                    "user": null,
                  }
                }
                loginUser={[Function]}
                submitForm={[Function]}
              />
            </div>
          </div>
        </div>
        <div
          className="col s1 m2 l4"
        />
      </div>,
      "className": "row wrap",
      "style": Object {
        "backgroundImage": "url(test-file-stub)",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="col s1 m2 l4 "
          />,
          <div
            className="col s10 m7 l5"
          >
            <div
              className="row card-panel"
            >
              <div
                className="row"
              >
                <ReduxForm
                  id="login-form"
                  login={
                    Object {
                      "isAuthenticated": false,
                      "isLoggingIn": false,
                      "user": null,
                    }
                  }
                  loginUser={[Function]}
                  submitForm={[Function]}
                />
              </div>
            </div>
          </div>,
          <div
            className="col s1 m2 l4"
          />,
        ],
        "className": "wrapper valign-wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "col s1 m2 l4 ",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="row card-panel"
            >
              <div
                className="row"
              >
                <ReduxForm
                  id="login-form"
                  login={
                    Object {
                      "isAuthenticated": false,
                      "isLoggingIn": false,
                      "user": null,
                    }
                  }
                  loginUser={[Function]}
                  submitForm={[Function]}
                />
              </div>
            </div>,
            "className": "col s10 m7 l5",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="row"
              >
                <ReduxForm
                  id="login-form"
                  login={
                    Object {
                      "isAuthenticated": false,
                      "isLoggingIn": false,
                      "user": null,
                    }
                  }
                  loginUser={[Function]}
                  submitForm={[Function]}
                />
              </div>,
              "className": "row card-panel",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <ReduxForm
                  id="login-form"
                  login={
                    Object {
                      "isAuthenticated": false,
                      "isLoggingIn": false,
                      "user": null,
                    }
                  }
                  loginUser={[Function]}
                  submitForm={[Function]}
                />,
                "className": "row",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "id": "login-form",
                  "login": Object {
                    "isAuthenticated": false,
                    "isLoggingIn": false,
                    "user": null,
                  },
                  "loginUser": [Function],
                  "submitForm": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "col s1 m2 l4",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="wrapper valign-wrapper"
        >
          <div
            className="col s1 m2 l4 "
          />
          <div
            className="col s10 m7 l5"
          >
            <div
              className="row card-panel"
            >
              <div
                className="row"
              >
                <ReduxForm
                  id="login-form"
                  login={
                    Object {
                      "isAuthenticated": false,
                      "isLoggingIn": false,
                      "user": null,
                    }
                  }
                  loginUser={[Function]}
                  submitForm={[Function]}
                />
              </div>
            </div>
          </div>
          <div
            className="col s1 m2 l4"
          />
        </div>,
        "className": "row wrap",
        "style": Object {
          "backgroundImage": "url(test-file-stub)",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="col s1 m2 l4 "
            />,
            <div
              className="col s10 m7 l5"
            >
              <div
                className="row card-panel"
              >
                <div
                  className="row"
                >
                  <ReduxForm
                    id="login-form"
                    login={
                      Object {
                        "isAuthenticated": false,
                        "isLoggingIn": false,
                        "user": null,
                      }
                    }
                    loginUser={[Function]}
                    submitForm={[Function]}
                  />
                </div>
              </div>
            </div>,
            <div
              className="col s1 m2 l4"
            />,
          ],
          "className": "wrapper valign-wrapper",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "col s1 m2 l4 ",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="row card-panel"
              >
                <div
                  className="row"
                >
                  <ReduxForm
                    id="login-form"
                    login={
                      Object {
                        "isAuthenticated": false,
                        "isLoggingIn": false,
                        "user": null,
                      }
                    }
                    loginUser={[Function]}
                    submitForm={[Function]}
                  />
                </div>
              </div>,
              "className": "col s10 m7 l5",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="row"
                >
                  <ReduxForm
                    id="login-form"
                    login={
                      Object {
                        "isAuthenticated": false,
                        "isLoggingIn": false,
                        "user": null,
                      }
                    }
                    loginUser={[Function]}
                    submitForm={[Function]}
                  />
                </div>,
                "className": "row card-panel",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <ReduxForm
                    id="login-form"
                    login={
                      Object {
                        "isAuthenticated": false,
                        "isLoggingIn": false,
                        "user": null,
                      }
                    }
                    loginUser={[Function]}
                    submitForm={[Function]}
                  />,
                  "className": "row",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "id": "login-form",
                    "login": Object {
                      "isAuthenticated": false,
                      "isLoggingIn": false,
                      "user": null,
                    },
                    "loginUser": [Function],
                    "submitForm": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
              "type": "div",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "col s1 m2 l4",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
